# Run the script as ' $ USER: clean_sample_reads.sh <PATH_TO_SAMPLE_LIST>'
# The <PATH_TO_SAMPLE_LIST> needs to be provided by the user and contains a list of "analyses" done
#	on samples, as produced by Pandora
# The folders in this Pandora list correspond to folder containing quality filtered reads 
#	that are mapped to the human genome (h19) by Kay using bwa version 0.7.12 (parameters: -n 0.01 -o 2 -l 16500). 
# These directory locations look as followed:
# /projects1/Autorun/Results/Human_Shotgun/180529_K00233_0080_AHMTWVBBXX/WHF001.C0101.SG1.1/

# Store the current path into a variable
CURRENT_DIR="$(pwd)"

# Check if Bam2FastQ is installed
for f in ~/Software/*; do
	export PATH=$PATH:$f
done

for f in ~/Software/*/bin/; do
	export PATH=$PATH:$f
done
# If not installed, install
command -v bam2fastq >/dev/null 2>&1 || { echo >&2 "Bam2fastq is not installed";
	echo "Installing now.";
	cd ~/Software/ ;
	wget https://gslweb.discoveryls.com/static/software/bam2fastq-1.1.0.tgz -q -o /dev/null --no-check-certificate;
	wait ;
	tar -xzf bam2fastq-1.1.0.tgz ;
	wait ;
	rm bam2fastq-1.1.0.tgz
	cd bam2fastq-1.1.0/ ;
	make ;
	wait ;
	export PATH=$PATH:~/Software/bam2fastq-1.1.0/ ;
	cd ${CURRENT_DIR} ;
}

# Get the command line arguments
SAMPLE_FOLDERS_LIST=$@

# Now run the script to extract names from an excell, 
if [ -f ${SAMPLE_FOLDERS_LIST} ]; then
	while read SAMPLE_FOLDER; do
		# Separate the csv file generated by Pandora by "," and store into an array
		IFS="," read -a SAMPLE_DATA_ARRAY <<< $SAMPLE_FOLDER
		# entry 1 is the name (column one of the csv file)
		SAMPLE_NAME="${SAMPLE_DATA_ARRAY[0]}"
		# entry 2 is the directory
		SAMPLE_LOCATION="${SAMPLE_DATA_ARRAY[1]}"
		# Remove the quotation marks around the string so we can directly use it
		sed -n -e 's/^"//'  -e 's/"$//' <<<"$SAMPLE_NAME"
		sed -n -e 's/^"//'  -e 's/"$//' <<<"$SAMPLE_LOCATION"
		echo "Extracting unmapped reads from sample: ${SAMPLE_NAME}"
		mkdir -p ${SAMPLE_NAME}
		echo "bam2fastq ${SAMPLE_LOCATION}out.bam -o ${SAMPLE_NAME}/${SAMPLE_NAME}_non-human_reads#.fastq --no-aligned --unaligned --no-filtered" | qsub -pe make 2 -l h_vmem=16G -e Malt_output/${SAMPLE_NAME}/${SAMPLE_NAME}_Malt_slurm_STDerr.txt -o Malt_output/${SAMPLE_NAME}/${SAMPLE_NAME}_Malt_slurm_STDout.txt
		-c 2 --mem=16000 --wrap=  --output=${SAMPLE_NAME}/${SAMPLE_NAME}_bam2fastq_stdout
	done < ${SAMPLE_FOLDERS_LIST}
else
	echo "The specified file does not appear to exist, please check the added path"
	exit 1;
fi
echo "Finished extracting reads"
